tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/table/TableNegativeTest.java

```diff
@@ -187,4 +187,17 @@ public void testTableKeyViolations() {
         validateError(compileResult, index, "duplicate key found in table row key('id') : 'ID2'",
                 136, 9);
     }
+
+    @Test
+    public void testAnyTypedTableWithKeySpecifiers() {
+        CompileResult compileResult = BCompileUtil.compile("test-src/types/table/table-value-any-negative.bal");
+        int index = 0;
+        validateError(compileResult, index++,
+                "field name 'name' used in key specifier is not found in table constraint type 'map<(any|error)>'",
```

Comment:
Shall we introduce a new error may be which addresses the info gap between how CEP `any` made the table constraint to be map<any|error>?

Reply:
`Key specifier not allowed when contextually expected type is any` ?

Reply:
maybe `Key specifier not allowed when the target type is any`